src/pages/registration/ContributorsPanel.tsx

```diff
@@ -1,28 +1,63 @@
-import { useFormikContext } from 'formik';
-import React from 'react';
+import { FormHelperText } from '@material-ui/core';
+import { ErrorMessage, FieldArray, FieldArrayRenderProps, useFormikContext } from 'formik';
+import React, { useEffect, useRef } from 'react';
+import BackgroundDiv from '../../components/BackgroundDiv';
+import lightTheme from '../../themes/lightTheme';
 import { ContributorRole } from '../../types/contributor.types';
-import { BookType, PublicationType } from '../../types/publicationFieldNames';
+import { BookType, ContributorFieldNames, PublicationType } from '../../types/publicationFieldNames';
 import { Registration } from '../../types/registration.types';
-import { Contributors } from './contributors_tab/components/Contributors';
+import { Contributors } from './contributors_tab/Contributors';
 
 const ContributorsPanel = () => {
   const {
     values: {
       entityDescription: {
         reference: { publicationContext, publicationInstance },
+        contributors,
       },
     },
+    errors,
+    touched,
+    setFieldValue,
   } = useFormikContext<Registration>();
 
-  return publicationContext.type === PublicationType.DEGREE ? (
+  const contributorsError = errors.entityDescription?.contributors;
+  const contributorsTouched = touched.entityDescription?.contributors;
+  const contributorsRef = useRef(contributors);
+
+  useEffect(() => {
+    // Ensure all contributors has a role by setting Creator role as default
+    const contributorsWithRole = contributorsRef.current.map((contributor) => ({
+      ...contributor,
+      role: contributor.role ?? ContributorRole.CREATOR,
+    }));
+    setFieldValue(ContributorFieldNames.CONTRIBUTORS, contributorsWithRole);
+  }, [setFieldValue]);
```

Comment:
denne er litt "bah", for det er bare for å håndtere gamle regisreringer fra før rolle ble satt. Egentlig skal det ikke være nødvendig, men trengs nå pga trøbbel med datamigrering :(

Reply:
får huske på å fjerne den etterhvert!